### PR TITLE
Add ScmInfo on build completion

### PR DESCRIPTION
**Pull request by lucamilanesio**
The build completion event can get the SCM Info from the
node environment variables and thus having a complete
record of what impacted which commit.

### CODE REVIEW

**Comment by lucamilanesio** [comment]
@maximecharron this PR is now ready for review :-) and has increased as well the test coverage with the adoption of the Jenkins Test Harness. Would you have time to review it? Thanks in advance. Luca.

File: pom.xml [hunk 1]
```diff
@@ -115,6 +115,12 @@
             <version>1.650</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.jenkins-ci.main</groupId>
+            <artifactId>jenkins-test-harness</artifactId>
+            <version>1.644</version>
```
Comment (lucamilanesio) [inline pom.xml]: There is no 1.650 version available for the jenkins-test-harness. The 1.644 however is good enough for our tests :-)

**Comment by lucamilanesio** [comment]
@alexgandy would you have some time to review my PR?

**Comment by lucamilanesio** [comment]
@alexgandy thanks for the approval, do you know if @maximecharron is still active on this project?

**Comment by alexgandy** [comment]
@lucamilanesio Unfortunately, I do not know the status of @maximecharron and forking the codebase to become the maintainer might be your best bet at quicker progress!

**Pull request merged**